### DISCUSSION TITLE
Is it necessary to print an error message when there is no socket available to accept in an asynchronous socket?

### DISCUSSION THREAD

**Discussion by xiaozhihong** [Haivision/srt#2768] [General]
This log is necessary?

https://github.com/Haivision/srt/blob/1737e96a0fa12fcd619b7818e959bef53b56a5b7/srtcore/api.cpp#L1133

**Comment by ethouris** [comment]
Yes. We have error log messages for every (unless we missed something) situation when an API call has ended up with error, so we can be aware of the real situation that caused the problem.

In this case it appears when you are calling `srt_accept` without checking first if there is a pending socket on the listener - although it might also happen in case when the pending socket was closed internally before connection completion.

**Comment by crazydef** [comment]
@ethouris For future readers, can you explain how to know when there's a pending connection or not because your [Listener (Server) Example](https://github.com/Haivision/srt/blob/master/docs/API/API.md#binding-and-connecting)  just calls `srt_accept` in a loop:

```
sockaddr_in sa = { ... }; // set local listening port and possibly interface's IP
int st = srt_bind(sock, (sockaddr*)&sa, sizeof sa);
srt_listen(sock, 5);
while ( !finish ) {
    int sa_len = sizeof sa;
    newsocket = srt_accept(sock, (sockaddr*)&sa, &sa_len);
    HandleNewClient(newsocket, sa);
}
```

If that's the only way to know if there's a pending connection, then it is unacceptable to constantly print an error message.
